docs/how-to/rest-api-and-dynamodb.md

```diff
@@ -0,0 +1,1364 @@
+
+```javascript
+// Example of web server written with Express! (much easier!)
+
+const express = require('express')
+const app = express()
+const port = 3000
+
+app.get('/', (req, res) => {
+  res.send('Hello World!')
+})
+
+app.listen(port, () => {
+  console.log(`Example app listening on port ${port}`)
+})
+```
+
+Thankfully, Node comes with its own package manager, giving us access to a plethora of community-made modules. One of those modules is Express, the most popular Node web framework. It comes with a variety of features out of the box:
+
+- Handling requests for the same URL with different HTTP verbs,
+- Parsing JSON data straight from HTTP requests,
+- Reading query data straight from HTTP requests,
+- Running “middleware” for any routes before handling the requests. (Useful for authentication)
+
+Express also comes with its own bundle of community-made middleware to handle all kinds of necessities such as parsing cookies and security headers. We’ll be using Express in the remainder of this tutorial.
+
+### Let’s understand RESTful APIs
+
+![REST APIs.png](../../static/img/tutorial/rest-api/REST_APIs.png)
+
+Server-side programmers have the ability to build their APIs however they want. That’s the freedom that Node and Express give you:
+
+- you get to name your own routes,
+- you get to handle different HTTP verbs in any way you want,
+- and you even get to format the responses in whichever way you desire.
+
+But with all this freedom comes a problem: we want all APIs to be unified under one standard; one way of formatting routes; one way of handling HTTP verbs; and one way of sending responses.
+
+That’s where REST comes in; it stands for “R**epresentational State Transfer**” and defines a universal standard for building APIs.
+
+REST recommends building your API around database(s), where each URL links in one way or another to one or more resources in that database.
+
+REST calls for the use of the following HTTP verbs:
+
+- **GET** is for fetching the resource.
+- **POST** is for creating the resource.
+- **DELETE** is for deleting the resource.
+- **PUT** is for updating the whole resource.
+- **PATCH** is for updating parts of the resource.
+
+> It’s important that **GET** requests (also known as [“Safe methods”](https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP)) do not change anything about the database.
+> 
+
+RESTful APIs handle routes in an “****[Idempotent](https://restfulapi.net/idempotent-rest-apis/)”** manner, meaning that making the same request multiple times yields the same result. They must also be [stateless](https://www.restapitutorial.com/lessons/whatisrest.html#), meaning that all necessary data to handle the request is contained within the request itself and not remembered from a previous request. This makes RESTful APIs much more scalable than their counterparts.
```

Comment:
Please check the formatting. I think this is one too many pairs of *'s

Also, can we replace all special quotes aka `“` and `”` with standard double-quotes? `"`